interop/xds/client/client.go

```diff
@@ -73,21 +73,71 @@ func (watcher *statsWatcher) buildResp() *testpb.LoadBalancerStatsResponse {
 	}
 }
 
+type accumulatedStats struct {
+	mu                       sync.Mutex
+	numRpcsStartedByMethod   map[string]int32
+	numRpcsSucceededByMethod map[string]int32
+	numRpcsFailedByMethod    map[string]int32
+}
+
+func copyStatsMap(originalMap map[string]int32) (newMap map[string]int32) {
+	newMap = make(map[string]int32)
+	for k, v := range originalMap {
+		newMap[k] = v
+	}
+	return newMap
+}
+
+func (as *accumulatedStats) buildResp() *testpb.LoadBalancerAccumulatedStatsResponse {
+	as.mu.Lock()
+	defer as.mu.Unlock()
+	return &testpb.LoadBalancerAccumulatedStatsResponse{
+		NumRpcsStartedByMethod:   copyStatsMap(as.numRpcsStartedByMethod),
+		NumRpcsSucceededByMethod: copyStatsMap(as.numRpcsSucceededByMethod),
+		NumRpcsFailedByMethod:    copyStatsMap(as.numRpcsFailedByMethod),
+	}
+}
+
+func (as *accumulatedStats) startRPC(rpcType string) {
+	as.mu.Lock()
+	defer as.mu.Unlock()
+	as.numRpcsStartedByMethod[rpcType]++
+}
+
+func (as *accumulatedStats) finishRPC(rpcType string, failed bool) {
+	as.mu.Lock()
+	defer as.mu.Unlock()
+	if failed {
+		as.numRpcsFailedByMethod[rpcType]++
+		return
+	}
+	as.numRpcsSucceededByMethod[rpcType]++
+}
+
 var (
 	failOnFailedRPC = flag.Bool("fail_on_failed_rpc", false, "Fail client if any RPCs fail after first success")
 	numChannels     = flag.Int("num_channels", 1, "Num of channels")
 	printResponse   = flag.Bool("print_response", false, "Write RPC response to stdout")
 	qps             = flag.Int("qps", 1, "QPS per channel, for each type of RPC")
-	rpc             = flag.String("rpc", "UnaryCall", "Types of RPCs to make, ',' separated string. RPCs can be EmptyCall or UnaryCall")
-	rpcMetadata     = flag.String("metadata", "", "The metadata to send with RPC, in format EmptyCall:key1:value1,UnaryCall:key2:value2")
+	rpc             = flag.String("rpc", "UnaryCall", "Types of RPCs to make, ',' separated string. RPCs can be EmptyCall or UnaryCall. Deprecated: Use Configure RPC to XdsUpdateClientConfigureServiceServer instead.")
+	rpcMetadata     = flag.String("metadata", "", "The metadata to send with RPC, in format EmptyCall:key1:value1,UnaryCall:key2:value2. Deprecated: Use Configure RPC to XdsUpdateClientConfigureServiceServer instead.")
 	rpcTimeout      = flag.Duration("rpc_timeout", 20*time.Second, "Per RPC timeout")
 	server          = flag.String("server", "localhost:8080", "Address of server to connect to")
 	statsPort       = flag.Int("stats_port", 8081, "Port to expose peer distribution stats service")
 
+	rpcCfgsMu sync.Mutex
+	rpcCfgs   = []*rpcConfig{{typ: unaryCall}}
```

Comment:
Since we don't care about consistency, you can make this an atomic value to save that Mutex.

Reply:
Made this change.